src/api/alchemy-websocket-provider.ts

```diff
@@ -675,6 +755,128 @@ export class AlchemyWebSocketProvider
         throw new Error('Invalid event type to `emitProcessFn()`');
     }
   }
+
```

Comment:
I am very much not a fan of all this copy/paste, but since `getEventTag()` isn't exposed, I can't really think of another way to support Alchemy's subscription APIs. @dphilipson any advice on how to clean this up?

Reply:
I don't know a good alternative unfortunately. Seems like we're stuck: we need to modify part of the function that isn't designed for extension, so nothing we can do other than replace the function entirely.